tools/kagami/CLI.md

```diff
@@ -0,0 +1,315 @@
+
+   ```bash
+   kagami genesis --inlined-validator
+   ```
+- Generate a genesis block in JSON format and write the output to the specified file:
+
+   ```bash
+   kagami genesis --inlined-validator >genesis.json
+   ```
+- Generate a synthetic genesis block in JSON format and write the `n` domains, `m` accounts per domain and `p` assets per domain:
+
+   ```bash
+   kagami genesis --inlined-validator synthetic --domains n --accounts-per-domain m --assets-per-domain p
+   ```
+
+- Generate a default genesis block in JSON format and provide path to the validator file (it could be absolute or relative to genesis file)
+
+   ```bash
+   kagami genesis --compiled-validator-path ./validator.wasm
+   ```
+
+**Usage:** `kagami genesis <--inlined-validator|--compiled-validator-path <COMPILED_VALIDATOR_PATH>> [COMMAND]`
+
+###### **Subcommands:**
+
+* `default` — Generate default genesis
+* `synthetic` — Generate synthetic genesis with specified number of domains, accounts and assets
+
+###### **Options:**
+
+* `--inlined-validator` — If this option provided validator will be inlined in the genesis
+* `--compiled-validator-path <COMPILED_VALIDATOR_PATH>` — If this option provided validator won't be included in the genesis and only path to the validator will be included. Path is either absolute path to validator or relative to genesis location. Validator can be generated using `kagami validator` command
+
+
+
+## `kagami genesis default`
+
+Generate default genesis
+
+**Usage:** `kagami genesis default`
+
+
+
+## `kagami genesis synthetic`
+
+Generate synthetic genesis with specified number of domains, accounts and assets.
+
+Synthetic mode is useful when we need a semi-realistic genesis for stress-testing Iroha's startup times as well as being able to just start an Iroha network and have instructions that represent a typical blockchain after migration.
+
+**Usage:** `kagami genesis synthetic [OPTIONS]`
+
+###### **Options:**
+
+* `--domains <DOMAINS>` — Number of domains in synthetic genesis
+
+  Default value: `0`
+* `--accounts-per-domain <ACCOUNTS_PER_DOMAIN>` — Number of accounts per domains in synthetic genesis. Total number of  accounts would be `domains * assets_per_domain`
+
+  Default value: `0`
+* `--assets-per-domain <ASSETS_PER_DOMAIN>` — Number of assets per domains in synthetic genesis. Total number of assets would be `domains * assets_per_domain`
+
+  Default value: `0`
+
+
+
+## `kagami config`
+
+Generate the default client/peer configuration
+
+**Usage:** `kagami config <COMMAND>`
+
+###### **Subcommands:**
+
+* `client` — 
+* `peer` — 
+
+
+
+## `kagami config client`
+
+**Usage:** `kagami config client`
+
+
+
+## `kagami config peer`
+
+**Usage:** `kagami config peer`
+
+
+
+## `kagami docs`
+
+Generate a Markdown reference of configuration parameters
+
+**Usage:** `kagami docs`
+
+
+
+## `kagami tokens`
+
+Generate a list of predefined permission tokens and their parameters
+
+**Usage:** `kagami tokens`
```

Comment:
The comment about `Args::Config` also applies here, despite this part being insubstantial. Even though it's just two small and obvious examples there, I don't think we lose much by omitting them, but being more explicit never hurts.